KitePrintSDK/src/main/java/ly/kite/analytics/Analytics.java

```diff
@@ -0,0 +1,660 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ *****************************************************/
+
+///// Package Declaration /////
+
+package ly.kite.analytics;
+
+
+///// Import(s) /////
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.util.DisplayMetrics;
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Currency;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.UUID;
+
+import ly.kite.BuildConfig;
+import ly.kite.KiteSDK;
+import ly.kite.address.Address;
+import ly.kite.address.Country;
+import ly.kite.print.PrintJob;
+import ly.kite.print.PrintOrder;
+import ly.kite.print.Product;
+
+
+///// Class Declaration /////
+
+/*****************************************************
+ *
+ * This class performs analytics processing and uploads
+ * data using the appropriate analytics agent.
+ *
+ *****************************************************/
+public class Analytics
+  {
+  ////////// Static Constant(s) //////////
+
+  @SuppressWarnings( "unused" )
+  private static final String  LOG_TAG                                     = "Analytics";
+
+  private static final String  SHARED_PREFRENCES_NAME                      = "kite_sdk_analytics_shared_prefs";
+  private static final String  SHARED_PREFERENCES_KEY_UNIQUE_USER_ID       = "unique_user_id";
+
+
+  private static final String  EVENT_NAME_SDK_LOADED                       = "Kite Loaded";
+  private static final String  EVENT_NAME_PRODUCT_SELECTION_SCREEN_VIEWED  = "Product Selection Screen Viewed";
+  private static final String  EVENT_NAME_PRODUCT_OVERVIEW_SCREEN_VIEWED   = "Product Description Screen Viewed";
+  private static final String  EVENT_NAME_CREATE_PRODUCT_SCREEN_VIEWED     = "Review Screen Viewed";
+  private static final String  EVENT_NAME_CREATE_PAYMENT_SCREEN_VIEWED     = "Payment Screen Viewed";
+  private static final String  EVENT_NAME_PAYMENT_COMPLETED                = "Payment Completed";
+  private static final String  EVENT_NAME_ORDER_SUBMISSION                 = "Print Order Submission";
+
+
+  // Some of these names originally come from the iOS SDK, and we need to match them,
+  // despite some not being entirely appropriate (e.g. bundle id) for Android.
```

Comment:
Feel free to use the Android equivalent for App Bundle Id -- "App Package Name"? Any others, at a glance that appears to be the only contentious one?